spec/std/range_spec.cr

```diff
@@ -80,23 +80,84 @@ describe "Range" do
   end
 
   it "does union" do
-    (1..5).union(3..7).should eq(1..7)
-    (1..10).union(1...15).should eq(1...15)
-    (1...10).union(1..15).should eq(1..15)
-    (1..5).union(10..20).should eq(nil)
-  end
-
+    # Int ranges
+    (1..5).union(3..7).should eq(1..7)          # Overlapping integer ranges
+    
+    # TODO: This test will fail - 
+    #   In an integer setting this case is doable. We can simply check adjaceny with +-1.
+    #   However, how do we define an adjacent range for other data types? 
+    #     Examples of adjacency ambiguity:
+    #       - Floats: Is 4.1's upper adjacent value 4.2 or 4.11, etc.?
+    #       - Times: Is a time's adjacent value +1 second or +1 hour, etc.?
+    # 
+    #   How do we define adjacency?
+    #   
+    # (1..5).union(6..10).should eq(1..10)      # Adjacent ranges
```

Comment:
As per the [Range documentation](https://crystal-lang.org/api/1.14.0/Range.html) you'd use `#succ` and `#pred`:

> Ranges typically involve integers, but can be created using arbitrary objects as long as they define succ (or pred for [#reverse_each](https://crystal-lang.org/api/1.14.0/Range.html#reverse_each%28%26%29%3ANil-instance-method)), to get the next element in the range, and < and [#==](https://crystal-lang.org/api/1.14.0/Range.html#%3D%3D%28other%3ARange%29-instance-method), to know when the range reached the end:

Reply:
> As per the [Range documentation](https://crystal-lang.org/api/1.14.0/Range.html) you'd use `#succ` and `#pred`:
> 
> > Ranges typically involve integers, but can be created using arbitrary objects as long as they define succ (or pred for [#reverse_each](https://crystal-lang.org/api/1.14.0/Range.html#reverse_each%28%26%29%3ANil-instance-method)), to get the next element in the range, and < and [#==](https://crystal-lang.org/api/1.14.0/Range.html#%3D%3D%28other%3ARange%29-instance-method), to know when the range reached the end:

Making this change this weekend. ✨